# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1643838899 -3600
# Parent  4db1b9451a01c5c0a60845413838c82a96ecbfd1
Feature: Add "Open Message For ID" to context menu in news message.

diff --git a/mail/base/content/mailContextMenus.js b/mail/base/content/mailContextMenus.js
--- a/mail/base/content/mailContextMenus.js
+++ b/mail/base/content/mailContextMenus.js
@@ -630,16 +630,20 @@ function addEmail() {
   window.openDialog(
     "chrome://messenger/content/addressbook/abNewCardDialog.xhtml",
     "",
     "chrome,resizable=no,titlebar,modal,centerscreen",
     { primaryEmail: addresses }
   );
 }
 
+function openMessageForMsgId() {
+  OpenMessageForMessageId(getEmail(gContextMenu.linkURL));
+}
+
 function composeEmailTo() {
   let fields = Cc[
     "@mozilla.org/messengercompose/composefields;1"
   ].createInstance(Ci.nsIMsgCompFields);
   let params = Cc[
     "@mozilla.org/messengercompose/composeparams;1"
   ].createInstance(Ci.nsIMsgComposeParams);
   fields.to = getEmail(gContextMenu.linkURL);
diff --git a/mail/base/content/mainPopupSet.inc.xhtml b/mail/base/content/mainPopupSet.inc.xhtml
--- a/mail/base/content/mainPopupSet.inc.xhtml
+++ b/mail/base/content/mainPopupSet.inc.xhtml
@@ -412,16 +412,21 @@
     <menuitem id="mailContext-copylink"
               label="&copyLinkCmd.label;"
               accesskey="&copyLinkCmd.accesskey;"
               command="cmd_copyLink"/>
     <menuitem id="mailContext-copyimage"
               label="&copyImageAllCmd.label;"
               accesskey="&copyImageAllCmd.accesskey;"
               command="cmd_copyImage"/>
+#ifndef BROWSER_POPUPS_ONLY
+    <menuitem id="mailContext-openMessageForMsgId"
+              label="&OpenMessageForMsgId.label;"
+              oncommand="openMessageForMsgId();"/>
+#endif
     <menuitem id="mailContext-addemail"
               label="&AddToAddressBook.label;"
               accesskey="&AddToAddressBook.accesskey;"
               oncommand="addEmail();"/>
     <menuitem id="mailContext-composeemailto"
               label="&SendMessageTo.label;"
               accesskey="&SendMessageTo.accesskey;"
               oncommand="composeEmailTo();"/>
diff --git a/mail/base/content/nsContextMenu.js b/mail/base/content/nsContextMenu.js
--- a/mail/base/content/nsContextMenu.js
+++ b/mail/base/content/nsContextMenu.js
@@ -415,16 +415,22 @@ class nsContextMenu {
       "mailContext-selectall",
       !this.inThreadPane && !this.onPlayableMedia
     );
     this.showItem("mailContext-copyemail", this.onMailtoLink);
     this.showItem("mailContext-copylink", this.onLink && !this.onMailtoLink);
     this.showItem("mailContext-copyimage", this.onImage);
 
     this.showItem(
+      "mailContext-openMessageForMsgId",
+      this.onMailtoLink &&
+        !this.inThreadPane &&
+        gFolderDisplay.selectedMessageIsNews
+    );
+    this.showItem(
       "mailContext-composeemailto",
       this.onMailtoLink && !this.inThreadPane
     );
     this.showItem(
       "mailContext-addemail",
       this.onMailtoLink && !this.inThreadPane
     );
 
